src/cfg.cpp

```diff
@@ -164,6 +179,25 @@ bool equal_user_mode(const void *value, const void *data) {
 	return true;
 }
 
+bool equal_user_transform(const void *value, const void *data) {
+	if (!value || !data) {
+		return false;
+	}
+
+	struct UserTransform *lhs = (struct UserTransform*)value;
+	struct UserTransform *rhs = (struct UserTransform*)data;
+
```

Comment:
We only need to test the values for equality. No need to look at names.

There is no such thing as a "null" transform, as the default `WL_OUTPUT_TRANSFORM_NORMAL` is 0.

Reply:
I realise I have made a huge mistake; we do need to test the names and values in this test as it is for the entire config.

Like `equal_user_mode`

```c
bool equal_user_transform(const void *value, const void *data) {
	if (!value || !data) {
		return false;
	}

	struct UserTransform *lhs = (struct UserTransform*)value;
	struct UserTransform *rhs = (struct UserTransform*)data;

	if (!lhs->name_desc || !rhs->name_desc) {
		return false;
	}

	if (strcasecmp(lhs->name_desc, rhs->name_desc) != 0) {
		return false;
	}

	if (lhs->transform != rhs->transform) {
		return false;
	}

	return true;
}
```